aya/src/obj/btf/btf.rs

```diff
@@ -156,6 +156,15 @@ pub enum BtfError {
     /// unable to get symbol name
     #[error("Unable to get symbol name")]
     InvalidSymbolName,
+
+    /// BTF Map parse error. Unexpected Type Encountered
+    #[error("map parse error. expected type {expected} but found type {found}")]
+    MapParseUnexpectedType {
```

Comment:
I think instead of this we should probably use the existing `UnexpectedBtfType` variant. I don't think it's an error that we're actually expecting to happen (except for corrupt files), so it's not worth adding an extra variant (API).